### PR TITLE
Fix typescript issues

### PR DESCRIPTION
**Pull request by juliankrispel**
The repository is currently broken for typescript users, this fixes it.

- [x] remove redundand package-lock (since there already is a yarn.lock it seems redundant)
- [x] update cryptopunks.ts to fix typescript errors
- [x] add tsconfig with assemblyscript defaults
- [x] update dependencies
- [ ] Update matchstick-as once it has upgraded it's dependencies ([there's already an open pr for that](https://github.com/LimeChain/matchstick-as/pull/36))

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
...
           git config --global url."https://github.com/".insteadOf
           ssh://git@github.com/
-    - run: npm ci
-    - run: npm run codegen
-    - run: npm run build
-    - run: npm run test
+    - run: yarn
+    - run: yarn codegen
+    - run: yarn build
+    - run: yarn test
```
Comment (juliankrispel) [inline .github/workflows/node.js.yml:38]: using `yarn` instead of `npm` so the package-lock can be gotten rid off.

Comment (georg-getz) [reply .github/workflows/node.js.yml:38]: Why would you want to remove package-lock? The reason we have them both is because some people prefer using npm and some prefer using yarn.

File: package.json [hunk 2]
```diff
@@ -8,12 +8,16 @@
     "build": "graph build",
     "create-local": "graph create example --node http://127.0.0.1:8020",
     "deploy-local": "graph deploy example --ipfs http://localhost:5001 --node http://127.0.0.1:8020",
-    "test": "graph test gravity"
+    "test": "tsc --traceResolution --noEmit --skipLibCheck && graph test gravity"
   },
   "dependencies": {
     "@graphprotocol/graph-cli": "^0.22.1",
     "@graphprotocol/graph-ts": "^0.22.1",
-    "matchstick-as": "git://github.com/LimeChain/matchstick-as.git#b9d6988a07923b8286d8c38ef25581368570bc2a",
```
Comment (georg-getz) [inline package.json:16]: We need that exact version of matchstick-as since some of its content is still not released and is therefore not in version 0.2.0

File: package.json [hunk 3]
```diff
@@ -8,12 +8,16 @@
...
     "keytar": "^7.4.0"
+  },
+  "devDependencies": {
+    "assemblyscript": "^0.19.20",
```
Comment (georg-getz) [inline package.json:19]: I don't think the assemblyscript dependency is necessary since we rely on libs which rely on assemblyscript

File: src/cryptopunks.ts [hunk 4]
```diff
@@ -415,9 +415,6 @@ export function handlePunkOffered(event: PunkOffered): void {
     );
   }
   ask.from = account.id;
-  if (!ask) {
-    ask.open = true;
-  }
```
Comment (georg-getz) [inline src/cryptopunks.ts:420]: Can you explain why this change is necessary? (I assume rows 516-518 are deleted for the same reason)

File: package.json (hunk 2 above)
Comment (georg-getz) [reply package.json:16]: Actually the commit hash we need is - #c2d83fd1f661cd5a773ef0b98d8c8a5d86c68a8e

File: src/cryptopunks.ts (hunk 4 above)
Comment (georg-getz) [reply src/cryptopunks.ts:420]: Actually we are going to delete the whole cryptopunks soon so I'm resolving this as it is not really relevant.

**Comment by georg-getz** [comment]
Closing this due to inactivity.
